lib/workload/stateful/token_service/token_service/cognitor/__init__.py

```diff
@@ -22,8 +22,14 @@ def __init__(self, user_pool_id: str, user_pool_app_client_id: str):
 
     @staticmethod
     def generate_password(length: int = 32) -> str:
+        """
+        Must meet the Cognito password requirements policy
+        https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-policies.html
+        """
+        if length < 8:
+            raise ValueError('Length must be at least 8 characters or more better')
         return ''.join(
-            random.SystemRandom().choice(string.ascii_letters + string.digits + '!@#$%&.') for _ in range(length)
+            random.SystemRandom().choice(string.ascii_letters + string.digits + '_-!@#%&.') for _ in range(length)
```

Comment:
Ah, user pool password policy demands at least one special character.... right!

Reply:
Yup. We can tune that though. But, let better be some symbols there, Flo.

Reply:
Yup, noticed the custom policy option. Agree though, better to stick to the default